R/1.0_module_data_summary.R

```diff
@@ -0,0 +1,159 @@
+        toggle_icon(session$ns("minimise_filter_overview"), c("fa-angle-right", "fa-angle-down"))
+        toggle_title(session$ns("minimise_filter_overview"), c("Restore panel", "Minimise Panel"))
+      })
+
+      output$table <- renderUI({
+        logger::log_trace("srv_data_summary updating counts")
+        if (length(active_datanames) == 0) {
+          return(NULL)
+        }
+
+        datasets_df <- get_filter_overview(datanames = active_datanames, datasets = datasets)
+
+        attr(datasets_df$dataname, "label") <- "Data Name"
+
+        if (!is.null(datasets_df$obs)) {
+          # some datasets (MAE colData) doesn't return obs column
+          datasets_df <- transform(
+            datasets_df,
+            obs_str_summary = ifelse(
+              !is.na(obs),
+              sprintf("%s/%s", obs_filtered, obs),
+              ""
+            )
+          )
+          attr(datasets_df$obs_str_summary, "label") <- "Obs"
+        }
+
+
+        if (!is.null(datasets_df$subjects)) {
+          # some datasets (without keys) doesn't return subjects
+          datasets_df <- transform(
+            datasets_df,
+            subjects_summary = ifelse(
+              !is.na(subjects),
+              sprintf("%s/%s", subjects_filtered, subjects),
+              ""
+            )
+          )
+          attr(datasets_df$subjects_summary, "label") <- "Subjects"
+        }
+
+        all_names <- c("dataname", "obs_str_summary", "subjects_summary")
+        datasets_df <- datasets_df[, colnames(datasets_df) %in% all_names]
+
+        body_html <- apply(
+          datasets_df,
+          1,
+          function(x) {
+            tags$tr(
+              tagList(
+                tags$td(
+                  if (all(x[-1] == "")) {
+                    icon(
+                      name = "exclamation-triangle",
+                      title = "Unsupported dataset",
+                      `data-container` = "body",
+                      `data-toggle` = "popover",
+                      `data-content` = "object not supported by the filter panel"
+                    )
+                  },
+                  x[1]
+                ),
+                lapply(x[-1], tags$td)
+              )
+            )
+          }
+        )
+
+        header_labels <- vapply(
+          seq_along(datasets_df),
+          function(i) {
+            label <- attr(datasets_df[[i]], "label")
+            ifelse(!is.null(label), label, names(datasets_df)[[i]])
+          },
+          character(1)
+        )
+        header_html <- tags$tr(tagList(lapply(header_labels, tags$td)))
+
+        table_html <- tags$table(
+          class = "table custom-table",
+          tags$thead(header_html),
+          tags$tbody(body_html)
+        )
+        logger::log_trace("srv_data_summary updated counts")
+        table_html
+      })
+      logger::log_trace("srv_data_summary initialized")
+      NULL
+    }
+  )
+}
+
+get_filter_overview = function(datanames, datasets) {
```

Comment:
Should be a dispatched s3 so that it also works for MAE

Reply:
Data will already be filtered as we are using `filtered_teal_data` 399cd9fe1217c15282072b76c7a9af5a8a9054b8